quipucords/tests/scanner/test_normalizer.py

```diff
@@ -0,0 +1,199 @@
+        "fact2": 42,
+    }
+    assert norm.metadata == {
+        "fact1": {
+            "has_error": False,
+            "raw_fact_keys": ["le_json"],
+            "server_id": server_id,
+            "source_type": N.source_type,
+        },
+        "fact2": {
+            "has_error": False,
+            "raw_fact_keys": ["le_json"],
+            "server_id": server_id,
+            "source_type": N.source_type,
+        },
+    }
+    # ensure data that would break a normalizer only affect one field
+    server_id2 = faker.uuid4()
+    norm2 = N({"le_json": '{"final_answer": "forty two"}'}, server_id2)
+    norm2.normalize()
+    assert norm2.facts == {
+        "fact1": {"final_answer": "forty two"},
+        "fact2": None,
+    }
+    assert norm2.metadata == {
+        "fact1": {
+            "has_error": False,
+            "raw_fact_keys": ["le_json"],
+            "server_id": server_id2,
+            "source_type": N.source_type,
+        },
+        "fact2": {
+            "has_error": True,
+            "raw_fact_keys": None,
+            "server_id": server_id2,
+            "source_type": N.source_type,
+        },
+    }
+
+
+class TestMultipleRawFacts:
+    """Test support for normalizers using multiple raw fact keys."""
+
+    @staticmethod
+    def _normalizer_simple(*, raw1, raw2):
+        return raw1 + raw2
+
+    @staticmethod
+    def _normalizer_custom_result(*, raw1, raw2):
+        value = raw1 + raw2
+        return NormalizedResult(value=value, raw_fact_keys=["raw1", "raw2"])
+
+    @staticmethod
+    def _normalizer_unknown_raw_fact(*, raw1, raw2):
+        value = raw1 + raw2
+        return NormalizedResult(value=value, raw_fact_keys=["raw3"])
+
+    @pytest.mark.parametrize(
+        "normalizer",
+        [
+            _normalizer_simple,
+            _normalizer_custom_result,
+        ],
+    )
+    def test_multiple_raw_facts(self, mocker, normalizer):
+        """Test normalization of facts that depend on multiple raw facts directly."""
+        raw_facts = {"raw1": 2, "raw2": 3}
+
+        class N(BaseNormalizer):
+            source_type = mocker.ANY
+            fact = FactMapper(["raw1", "raw2"], normalizer)
+
+        norm = N(raw_facts, mocker.ANY)
+        norm.normalize()
+        assert norm.facts == {"fact": 5}
+
+    def test_normalizer_with_unknown_raw_fact_key(self, mocker, caplog):
+        """Test a normalizer that says it depends on unexpected keys."""
+        raw_facts = {"raw1": 2, "raw2": 3}
+        caplog.set_level(logging.ERROR)
+
+        class N(BaseNormalizer):
+            source_type = mocker.ANY
+            fact = FactMapper(["raw1", "raw2"], self._normalizer_unknown_raw_fact)
+
+        norm = N(raw_facts, mocker.ANY)
+        norm.normalize()
+        assert norm.facts == {"fact": 5}
+        assert (
+            caplog.messages[-1] == "Unexpected raw facts used for fact 'fact': {'raw3'}"
+        )
+
+
+def test_multiple_normalizers():
```

Comment:
Would it be worth it to add some more cases in this vein?

- two normalizers, but with the same fact name
- one normalizer claiming a fact dependency on a fact that is only defined in another normalizer
- two normalizers inheriting from common parent, but other than BaseNormalizer (e.g. BaseNormalizer -> NetworkNormalizer -> N1 ; BaseNormalizer -> NetworkNormalizer -> N2). Assuming we expect to ever have normalizers inheriting indirectly from BaseNormalizer.